patroni/postgresql.py

```diff
@@ -100,9 +100,13 @@ def __init__(self, config):
         self._postgresql_conf = os.path.join(self._data_dir, config_base_name + '.conf')
         self._postgresql_base_conf_name = config_base_name + '.base.conf'
         self._postgresql_base_conf = os.path.join(self._data_dir, self._postgresql_base_conf_name)
+        self._postgresql_custom_conf = config.get('custom_conf')
         self._recovery_conf = os.path.join(self._data_dir, 'recovery.conf')
-        self._configuration_to_save = (self._postgresql_conf, self._postgresql_base_conf,
-                                       os.path.join(self._data_dir, 'pg_hba.conf'))
+        self._configuration_to_save = [self._postgresql_conf]
+        if not self._postgresql_custom_conf:
```

Comment:
> Also, when being restored on another instance that also has custom_conf and hba_file, what do we want to do? overwrite the file at the custom_conf path with the conf that was carried over, or copy it as postgresql.base.conf but keep it unused?.

The second option seems reasonable to be, but taking into account following:
> I find it cleaner to just explain: if you use custom_conf or hba_file, they will be under your control, there will not be any postgresql.base.conf and the configuration files will not be carried over on base_backups, and it's your responsability to ensure the settings on them are consistent and the files exist on all nodes.

I think you are right. We need document this behavior and let user live with it. Actually this is not the only drawback. Also I would like to add: currently Patroni is  "tracking" dynamic changes of postgres parameters and tells you in the API that restart is needed to apply some of the changes. In case of external configuration file it wont tell you this information..